examples/browser/Dockerfile.dev

```diff
@@ -0,0 +1,22 @@
+#
+# Build image:
+#    docker build -t theia-dev -f Dockerfile.dev .
+#
+# How to use this image
+#
+# theia-dev image helps build and run theia without installing anything on the host
+# 1.  Build theia in docker:
+#     docker run -e GITHUB_TOKEN="<token>" -v /path/to/theia-ide-sources-folder/:/home/theia theia-dev sh -c 'cd /home/theia; yarn'
+#
+#     Note that sometimes you can reach Github API rate limit, so to solve this issue env var GITHUB_TOKEN could be used as per example
+#
+# 2.  Run theia sources in docker:
+#     docker run -d -p 3000:3000 -v /path/to/theia-ide-sources-folder/:/home/theia theia-dev sh -c 'cd /home/theia/examples/browser; yarn run start --hostname 0.0.0.0'
+#
+
+FROM node:8
+RUN useradd --create-home theia
+WORKDIR /home/theia
+RUN rm -rf /opt/yarn && rm -f /usr/local/bin/yarn && rm -f /usr/local/bin/yarnpkg
```

Comment:
I think lines 20 and 21 can be removed, `node:8` contains yarn 1.3.2 by default: https://github.com/nodejs/docker-node/blob/master/8/Dockerfile#L43

and the version can be configured by setting the env variable

Reply:
make sense, I keep only git install here afaik theia need git client.
btw is there any reason why JDK is not included in that image by default? 
if this due to licensing stuff we could add openJDK

Reply:
I do not think it's about licensing. For sure JAVA is important, but one can use Theia for many other languages, without needing the JDK. So I think it's valid to have base image(s) without such a specialized tool. 

FYI, we have a separate [repo](https://github.com/theia-ide/theia-apps) for Theia applications in the form of Docker images. There we store DockerFiles for different use cases, [some of which](https://github.com/theia-ide/theia-apps/blob/master/theia-full-docker/Dockerfile#L95) also contain language tools and dependencies, such as the JDK. 

The one we have so far are meant for consumption, more than development, and so are not build from source, but from npm. Maybe we could add development images in there as well? These here being a start, then maybe more specialized images that add tools/language support for specific languages? i.e "theia-dev-java" ?

Reply:
@JanKoehnlein do you have an opinion about adding development images, with Theia build from source, to theia-apps?

Reply:
I am in favor of keeping docker files used for development in the separate place from the browser example. I am fine with using theia-apps for it. @svenefftinge what do you think?

Also:
- it is important that developer should be able to run Electron to check how their changes affect it. Would it be possible to setup docker images allowing it?
- developing in docker should remain optional, one should be able to start without installing docker

Reply:
I opened an issue, to discuss this in a way that can be better preserved for future reference. 
https://github.com/theia-ide/theia/issues/1153